torch/_tensor_str.py

```diff
@@ -527,11 +527,26 @@ def _str_intern(inp, *, tensor_contents=None):
             def indented_str(s, indent):
                 return "\n".join(f"  {line}" for line in s.split("\n"))
 
-            strs = ",\n".join(
-                indented_str(str(t), indent + 1)
-                for t in torch.ops.aten.unbind.int(self, 0)
-            )
-            tensor_str = f"[\n{strs}\n]"
+            if self.is_meta:
+                nt_sizes = self._nested_tensor_size()
+                suffixes.append(
+                    "num_components="
```

Comment:
Is `num_components` actually a valid kwarg to the constructor? (In general, it is best for the printing function to hew to the actual working syntax as much as possible.)

Reply:
> Is num_components actually a valid kwarg to the constructor?

Sadly, no :( The serialization here is made up and doesn't fit the proper semantics for `__repr__()`.

We don't have many public APIs for NT creation at the moment, so almost nothing we do here will actually work.